actors/miner/tests/miner_actor_test_wpost.rs

```diff
@@ -663,28 +671,570 @@ fn duplicate_proof_rejected_with_many_partitions() {
+    };
+    rt.expect_send(
+        *REWARD_ACTOR_ADDR,
+        reward::Method::ThisEpochReward as u64,
+        RawBytes::default(),
+        TokenAmount::from(0),
+        RawBytes::serialize(current_reward).unwrap(),
+        ExitCode::Ok,
+    );
+
+    let network_power = BigInt::from(1i64 << 50);
+    let current_power = power::CurrentTotalPowerReturn {
+        raw_byte_power: network_power.clone(),
+        quality_adj_power: network_power.clone(),
+        pledge_collateral: h.network_pledge,
+        quality_adj_power_smoothed: h.epoch_qa_power_smooth,
+    };
+    rt.expect_send(
+        *STORAGE_POWER_ACTOR_ADDR,
+        power::Method::CurrentTotalPower as u64,
+        RawBytes::default(),
+        TokenAmount::from(0),
+        RawBytes::serialize(current_power).unwrap(),
+        ExitCode::Ok,
+    );
+
+    let result = rt.call::<miner::Actor>(
+        miner::Method::DisputeWindowedPoSt as u64,
+        &RawBytes::serialize(params).unwrap(),
+    );
+    expect_abort_contains_message(
+        ExitCode::ErrForbidden,
+        "can only dispute window posts during the dispute window",
+        result,
+    );
+    rt.verify();
+}
 
 #[test]
-fn cant_dispute_up_with_an_invalid_deadline() {}
+fn cant_dispute_up_with_an_invalid_deadline() {
+    let period_offset = ChainEpoch::from(100);
+    let precommit_epoch = ChainEpoch::from(1);
+
+    let mut h = ActorHarness::new(period_offset);
+    h.set_proof_type(RegisteredSealProof::StackedDRG2KiBV1P1);
+
+    let mut rt = h.new_runtime();
+    rt.epoch = precommit_epoch;
+    rt.balance.replace(TokenAmount::from(BIG_BALANCE));
+
+    h.construct_and_verify(&mut rt);
+
+    let params = miner::DisputeWindowedPoStParams { deadline: 50, post_index: 0 };
+
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, h.worker);
+    rt.expect_validate_caller_type(vec![*ACCOUNT_ACTOR_CODE_ID, *MULTISIG_ACTOR_CODE_ID]);
+
+    let result = rt.call::<miner::Actor>(
+        miner::Method::DisputeWindowedPoSt as u64,
+        &RawBytes::serialize(params).unwrap(),
+    );
+    expect_abort_contains_message(ExitCode::ErrIllegalArgument, "invalid deadline", result);
+    rt.verify();
+}
 
 #[test]
-fn can_dispute_test_after_proving_period_changes() {}
+fn can_dispute_test_after_proving_period_changes() {
+    let period_offset = ChainEpoch::from(100);
+    let precommit_epoch = ChainEpoch::from(1);
+
+    let mut h = ActorHarness::new(period_offset);
+    h.set_proof_type(RegisteredSealProof::StackedDRG2KiBV1P1);
+
+    let mut rt = h.new_runtime();
+    rt.epoch = precommit_epoch;
+    rt.balance.replace(TokenAmount::from(BIG_BALANCE));
+
+    h.construct_and_verify(&mut rt);
+
+    let period_start = h.deadline(&rt).next_period_start();
+
+    // go to the next deadline 0
+    rt.epoch = period_start;
+
+    // fill one partition in each mutable deadline.
+    let num_sectors = h.partition_size * (rt.policy.wpost_period_deadlines - 2);
+
+    // creates a partition in every deadline except 0 and 47
```

Comment:
By the time this PR merges this should be true because I think its needed to get the first test to pass without skipping the fund checks, but you should double check that these are the actual deadlines in case you fix the first test without getting an exact match with the go deadline index.